### PR TITLE
Construct Arf/Arb/Arc from BigInt

### PR DESCRIPTION
**Pull request by lucifer1004**
solves #134

### CODE REVIEW

File: src/constructors.jl [hunk 1]
```diff
@@ -8,12 +8,14 @@ Arf(x; prec::Integer = _precision(x)) = set!(Arf(prec = prec), x)
...
 Arf(x::Arf; prec::Integer = precision(x)) = set!(Arf(prec = prec), x)
 Arf(x::Rational; prec::Integer = _precision(x)) = set!(Arf(prec = prec), x)
+Arf(x::BigInt; prec::Integer = precision(Arf)) = set_mpz!(Arf(prec = prec), x)
 
 #Arb
 Arb(x; prec::Integer = _precision(x)) = set!(Arb(prec = prec), x)
 # disambiguation
 Arb(x::Arb; prec::Integer = precision(x)) = set!(Arb(prec = prec), x)
 Arb(x::Rational; prec::Integer = _precision(x)) = set!(Arb(prec = prec), x)
+Arb(x::BigInt; prec::Integer = precision(Arb)) = Arb(Arf(x; prec = prec); prec = prec)
```
Comment (kalmarek) [inline src/constructors.jl:18]: I'd rather have to as
```julia
set!(Arb(prec = prec), Arf(x, prec = prec))
```
to keep in line of using `set!`-ters. I know this is not exactly the julia way, but that's what it is on the `arb` level

File: src/constructors.jl [hunk 2]
```diff
@@ -32,6 +34,8 @@ Acb(
     im::Union{Real,arb_struct,arf_struct,Tuple{<:Real,<:Real}};
     prec::Integer = max(_precision(re), _precision(im)),
 ) = set!(Acb(prec = prec), re, im)
+Acb(re::BigInt, im::BigInt; prec::Integer = precision(Acb)) =
+    Acb(Arb(re; prec = prec), Arb(im; prec = prec); prec = prec)
```
Comment (kalmarek) [inline src/constructors.jl:38]: ditto here: `set!(Acb(prec=prec), re, im)` would be preferred

**Review by kalmarek — CHANGES_REQUESTED** [review]

**Comment by kalmarek** [comment]
@lucifer1004 could you rebase on master?

**Comment by lucifer1004** [comment]
Rebased.

**Comment by Joel-Dahne** [comment]
I had completely missed that there is a `arf_set_mpz` method in Arb, that makes everything a lot easier since we don't have to go through `fmpz_t`! I think this will make things even easier actually, let me take a look at it!

**Comment by Joel-Dahne** [comment]
Okay so I took a look at this! The main reason this has not been working properly is that the `_mpz` suffix of the method name had not been striped in the Julia name.  So what needs to be fixed is actually our parsing code. Since you might not be that familiar with it it might be easier if I do it. Let me put something together and make another pull request and see if we get it to work there!

**Pull request closed without merging**